service/header/header_verify.go

```diff
@@ -0,0 +1,106 @@
+package header
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/tendermint/tendermint/light"
+)
+
+// TrustingPeriod is period through which we can trust a header's validators set.
+//
+// Should be significantly less than the unbonding period (e.g. unbonding
+// period = 3 weeks, trusting period = 2 weeks).
+//
+// More specifically, trusting period + time needed to check headers + time
+// needed to report and punish misbehavior should be less than the unbonding
+// period.
+// TODO(@Wondertan): We should request it from the network's state params
+//  or listen for network params changes to always have a topical value.
+var TrustingPeriod = 168 * time.Hour
```

Comment:
Can you explain the use case for a light client? I don't see a point to allow users to change this to unsafe values. Also, look at my TODO which says that we need to fetch it from network-wide params to be always consistent and thus safe.

Reply:
The use-case is that you can locally set to a stricter trusting period. Why would the network-wide param change once it is set?

Reply:
Do we need this in this PR?
>The use-case is that you can locally set to a stricter trusting period.
Not sure exactly, but I assume there are no fixed params for the network and everything can be changed during a chain is live. Or do you say you can change unbonding period only by regenesis?
> Why would the network-wide param change once it is set?

Reply:
> Do we need this in this PR?

No. But we should capture this in an issue.

> Not sure exactly, but I assume there are no fixed params for the network and everything can be changed during a chain is live. Or do you say you can change unbonding period only by regenesis?

There are also local params per light node. The trusting_period could be safely set to anything < unbondin_period locally. There will be a global unbonding period but that is unlikely to change (and it would require an upgrade).

Reply:
Ok: https://github.com/celestiaorg/celestia-node/issues/401
> No. But we should capture this in an issue.

Sure, I just don't see why this is a use case users would want. The fact that Tedermint does this, does not tell me why. I don't see this case giving anything to a user from a security standpoint, but only problems as user can harm himself by highering the value without even knowing what it is. Just look at russian th chat, people aped on every config field.
> The trusting_period could be safely set to anything < unbondin_period locally. 

How those upgrades are done? By that TODO I meant that It won't be safe for users to trust TrustingPeriod constant, as the unbonding period might be changed to even lower values by some upgrade system which does not affect the code. It is optimistic to think that we won't forget to update all such values in code after the update and it is even more optimistic to think that all the users will update instantly to the code with new values, thus runtime fetching/listening makes sense. This vision is against user facing configurations for security params, where we give users an ability to shoot into their own legs. However, that's far from now and I am just explaining my POV and why i don't see user based configuration for TrustingPeriod a valid direction, but until automatic approach, it might make sense to allow changing this param from config. 
> and it would require an upgrade